ONSdigital/sds
# Sdss 829 sds   mark datasets for deletion

### Motivation and Context

A method to mark datasets for collections which have ended for deletion since they are no longer required to be stored and take up date and following data retention policy.

### What has changed



Wrote an endpoint thats able to get Pub/Sub messages from a push subscription. (this is intended to replace the listener in a thread implemented in https://github.com/ONSdigital/sds/pull/161 ) 
The endpoint accepts a dataclass which can be modified when we have the concrete object RAS intend to use. 
The message recieved is checked to see if the 'dataset_id' is populated (we will also pick up messages from non prepop collections ending).
If there is preprop data then we attempt to collect all metadata associated with the survey_id and period.
These datasets are them marked to be deleted in a separate collection. 


### How to test?



In order to test:
1. Create a dataset
2. Place message onto topic (under Messages -> publish message)
3. Should have records for the datasets in 'marked_for_deletion'

### Links



https://jira.ons.gov.uk/browse/SDSS-829
https://github.com/ONSdigital/sds-iac/pull/131  IAC terraform changes 

### Screenshots (if appropriate):
